src/sysupdate/sysupdate.c

```diff
@@ -810,20 +835,24 @@ static int context_apply(
                 sync();
 
         (void) sd_notifyf(false,
-                          "STATUS=Updating to '%s'.\n", us->version);
+                          "STATUS=Updating to '%s'.", us->version);
 
         /* There should now be one instance picked for each transfer, and the order is the same */
         assert(us->n_instances == c->n_transfers);
 
         for (size_t i = 0; i < c->n_transfers; i++) {
-                r = transfer_acquire_instance(c->transfers[i], us->instances[i]);
+                r = transfer_acquire_instance(c->transfers[i], us->instances[i],
+                                              context_on_acquire_progress, c);
                 if (r < 0)
                         return r;
         }
 
         if (arg_sync)
                 sync();
 
+        (void) sd_notifyf(false,
```

Comment:
btw, our current coding style suggests prefixing "opaque" arguments with the parameter name in a comment i.e. arguments such as "0", "false" or "NULL" we typically want to be prefixed with something like `/* unset= */` in this case